x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_list_page/index.tsx

```diff
@@ -341,9 +341,10 @@ export const AgentListPage: React.FunctionComponent<{}> = () => {
 
   const isAgentSelectable = (agent: Agent) => {
     if (!agent.active) return false;
+    if (!agent.policy_id) return true;
 
-    const agentPolicy = agentPolicies.find((p) => p.id === agent.policy_id);
-    const isManaged = agent.policy_id && agentPolicy?.is_managed === true;
+    const agentPolicy = agentPoliciesIndexedById[agent.policy_id];
```

Comment:
@nchaulet This is the follow up mentioned in https://github.com/elastic/kibana/pull/96160#discussion_r607929896